### PR TITLE
added augmentation #1 (synonym replacement aug)

### PR DESCRIPTION
**Pull request by AmashiNiw**

### CODE REVIEW

File: src/conf/augmenter/synonym.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+_target_: augment.synonym.ApplySynonymAug
+l1: ${data.l1}
+l2: ${data.l2}
+lang1: null
+lang2: eng
```
Comment (adambarla) [inline src/conf/augmenter/synonym.yaml:5]: why are two definitions of language needed? (e.g. `l1` and `lang1`) ?

File: src/conf/main.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 defaults:
     - _self_
     - tokenizer: bpe
+    - augmenter: no_aug
```
Comment (adambarla) [inline src/conf/main.yaml]: wouldn't this work better with defualt being `augmenter: null`?

File: src/training.py [hunk 3]
```diff
@@ -98,8 +98,12 @@ def main(cfg):
         cfg.tokenizer, dataset=dataset, lang=cfg.data.l2
     )
     print(f"Tokenizer {cfg.data.l2}:\n{tokenizer_l2}")
+    augmenter = None
+    if not (cfg.augmenter == {}):
```
Comment (adambarla) [inline src/training.py]: if `augmenter: null` you could check 
```python
if cfg.augmenter is not None
```

**Review by adambarla — CHANGES_REQUESTED** [review]

File: src/training.py (hunk 3 above)
Comment (adambarla) [reply src/training.py]: you can get rid of the empty file then

File: src/conf/augmenter/synonym.yaml (hunk 1 above)
Comment (AmashiNiw) [reply src/conf/augmenter/synonym.yaml:5]: Wordnet (where I'm getting synonym data from) uses different codes for languages. This is a lazy fix :)
It also allows me to turn augmentations off for individual languages if needed.

File: src/conf/main.yaml (hunk 2 above)
Comment (AmashiNiw) [reply src/conf/main.yaml]: I tried, and it doesn't work. Doesn't pass None, simply ceases to exit. This was the fix I could do easily (first time user of hydra here!). There could be a more elegant fix

File: src/training.py (hunk 3 above)
Comment (AmashiNiw) [reply src/training.py]: That was my first implementation. Didn't work unfortunately.

**Comment by adambarla** [comment]
are you getting this?

```
'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
/Users/adam/miniconda3/envs/aug/lib/python3.10/site-packages/datasets/table.py:1283: FutureWarning: promote has been superseded by promote_options='default'.
  return cls._concat_blocks(pa_tables_to_concat_vertically, axis=0)
```

**Comment by AmashiNiw** [comment]
No. Not for cs-en.

**Review by adambarla — APPROVED** [review]

**Pull request merged**